### PR TITLE
Add ab testing for Go to Def (fix some ab testing bugs)

### PR DESCRIPTION
**Pull request by sean-mcmanus**

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -400,7 +402,8 @@ class DefaultClient implements Client {
                 default: {
                     systemIncludePath: settings.defaultSystemIncludePath
                 },
-                vcpkg_root: util.getVcpkgRoot()
+                vcpkg_root: util.getVcpkgRoot(),
+                goto_def_intelliSense: abTestSettings.UseGoToDefIntelliSense
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts]: please rename to `gotoDefIntelliSense` or similar.

File: Extension/src/LanguageServer/persistentState.ts [hunk 2]
```diff
@@ -20,7 +20,11 @@ class PersistentStateBase<T> {
     }
 
     public get Value(): T {
-         return this.state.get<T>(this.key, this.defaultvalue);
+        return this.state.get<T>(this.key, this.defaultvalue);
+    }
+
+    public get DefaultValue(): T {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/persistentState.ts]: nit: Can you move this under the setter for `Value` to keep the two properties together?

File: Extension/src/LanguageServer/client.ts (hunk 1 above)
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts]: Doing this in init works for now, but I'd imagine that at some point in the future we'll want a separate `abTestingSettingsChanged` message or something similar in the future.  That will allow us to make the settings read async again too.

**Review by bobbrow — APPROVED** [review]

File: Extension/src/LanguageServer/persistentState.ts (hunk 2 above)
Comment (sean-mcmanus) [reply Extension/src/LanguageServer/persistentState.ts]: Ok

**Pull request merged**